sql/api/src/main/scala/org/apache/spark/sql/catalyst/encoders/RowEncoder.scala

```diff
@@ -106,7 +106,7 @@ object RowEncoder extends DataTypeErrorsBase {
       UDTEncoder(udt, udtClass.asInstanceOf[Class[_ <: UserDefinedType[_]]])
     case ArrayType(elementType, containsNull) =>
       IterableEncoder(
-        classTag[mutable.ArraySeq[_]],
+        classTag[immutable.ArraySeq[_]],
```

Comment:
https://github.com/apache/spark/blob/af70aafd330fdbb6ce0d5b3efbcb180cda488695/sql/api/src/main/scala/org/apache/spark/sql/Row.scala#L317-L325

```
  /**
   * Returns the value at position i of array type as a Scala Seq.
   *
   * @throws ClassCastException when data type does not match.
   */
  def getSeq[T](i: Int): Seq[T] = {
    val res = getAs[scala.collection.Seq[T]](i)
    if (res != null) res.toSeq else null
  }
```

The change is to avoid the frequent collection copying when calling `getSeq`(`mutable.Seq` -> `immutable.Seq`), which can be costly especially when the collection itself is large."

Reply:
But if the user is used to using `row.getAs[mutable.Seq]` instead of `row.getSeq`, then he/she will have to switch to using `row.getAs[immutable.Seq]`.